src/middlewared/middlewared/plugins/failover_/virtual_ips.py

```diff
@@ -9,31 +8,45 @@ class Config:
         namespace = 'failover.vip'
 
     async def check_failover_group(self, ifname, groups):
-
         """
         Check the other members (if any) in failover group for `ifname`
         """
-
-        masters, backups = [], []
-
-        # get failover group id for `iface`
-        group_id = [group for group, names in groups.items() if ifname in names][0]
-
-        # get all interfaces in `group_id`
-        ids = [names for group, names in groups.items() if group == group_id][0]
-
-        # need to remove the passed in `ifname` from the list
-        ids.remove(ifname)
-
-        # we can have more than one interface in the failover
-        # group so check the state of the interface
-        for i in ids:
-            iface = netif.get_interface(i)
-            for j in iface.vrrp_config:
-                if j['state'] == 'MASTER':
-                    masters.append(i)
+        failover_grp_ifaces = list()
+        for grp, names in groups.items():
+            if ifname in names:
+                # get the list of interfaces that are in the same
+                # failover group as `ifname`.
+                failover_grp_ifaces = names
+
+                # we remove `ifname` since we only care about the other
+                # interfaces in this failover group
+                failover_grp_ifaces.remove(ifname)
```

Comment:
You are modifying the method input parameter here. This can produce very unexpected bugs (like the last time when some global list was growing indefinitely). It'll be safer to modify a copy.

Reply:
It doesn't matter in this instance since the object has a finite lifetime and is only generated on a failover event. However, I'll convert it to an immutable type.

Reply:
@themylogin thanks for review, just pushed a fix.